runtime/drivers/clickhouse/clickhouse.go

```diff
@@ -124,6 +127,11 @@ type configProperties struct {
 	ConnMaxLifetime string `mapstructure:"conn_max_lifetime"`
 	// ReadTimeout is the maximum amount of time a connection may be reused. Default is 300s.
 	ReadTimeout string `mapstructure:"read_timeout"`
+
+	// AWS S3 related configuration
+	AWSRoleARN                string `mapstructure:"aws_role_arn"`
+	AWSRoleSessionName        string `mapstructure:"aws_role_session_name"`
+	AWSSessionDurationSeconds *int32 `mapstructure:"aws_session_duration_seconds"`
```

Comment:
1. It feels wrong for these properties to be configured on the Clickhouse connector directly. Shouldn't they be implemented on the S3 connector (i.e. `runtime/drivers/s3.ConfigProperties{}`)? And then perhaps accessed from the Clickhouse driver in the `objectStoreToSelfExecutor`?
2. Why does Clickhouse have a different set of config properties for this than in the DuckDB driver?

Reply:
I missed the s3 driver on first pass. Agreed that makes the most sense.